### PR TITLE
Document policy for python version support.

### LINKED ISSUE #3347: Process: follow numpy's python support cadence
**Is your feature request related to a use case or problem? Please describe.**

We need clarity around what python versions we support. As python evolves, we get issues for incompatibility. (see issues labeled with python3.8), and we should have CI tests for the supported python versions. 

**Describe the solution you'd like**

As we depend on NumPy, we could follow their support process of 42 months window: https://numpy.org/neps/nep-0029-deprecation_policy.html

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P2 - we should do it in the next couple of quarters

### PR DESCRIPTION
**Pull request by maffoo**
Fixes #3347

### CODE REVIEW

File: docs/start/install.md [hunk 1]
```diff
@@ -17,6 +17,14 @@ We may change or remove parts of Cirq's API when making new releases.
 To be informed of deprecations and breaking changes, subscribe to the
 [cirq-announce google group mailing list](https://groups.google.com/forum/#!forum/cirq-announce).
 
+## Python version supprt
+
+Cirq currently supports python 3.7 and later.
+We follow numpy's schedule for python version support defined in [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html),
```
Comment (vtomole) [inline docs/start/install.md:23]:
```suggestion
We follow numpy's schedule for python version support defined in [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html);
```

File: docs/start/install.md [hunk 2]
```diff
@@ -17,6 +17,14 @@ We may change or remove parts of Cirq's API when making new releases.
...
+though we may deviate from that schedule by extending support for older versions,
+if needed for compatibility with [Colab](https://colab.research.google.com/)
```
Comment (vtomole) [inline docs/start/install.md:25]:
```suggestion
if we need to be compatible with [Colab](https://colab.research.google.com/)
```

File: docs/start/install.md [hunk 3]
```diff
@@ -17,6 +17,14 @@ We may change or remove parts of Cirq's API when making new releases.
 To be informed of deprecations and breaking changes, subscribe to the
 [cirq-announce google group mailing list](https://groups.google.com/forum/#!forum/cirq-announce).
 
+## Python version supprt
+
+Cirq currently supports python 3.7 and later.
+We follow numpy's schedule for python version support defined in [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html),
+though we may deviate from that schedule by extending support for older versions,
```
Comment (vtomole) [inline docs/start/install.md:24]:
```suggestion
though we may deviate from that schedule by extending support for older versions
```

**Review by vtomole — APPROVED** [review]
LGTM. Feel free to apply whatever syntax suggestions you agree with.

File: docs/start/install.md (hunk 1 above)
Comment (maffoo) [reply docs/start/install.md:23]: changed.

File: docs/start/install.md (hunk 3 above)
Comment (maffoo) [reply docs/start/install.md]: changed.

File: docs/start/install.md (hunk 2 above)
Comment (maffoo) [reply docs/start/install.md]: reworded.

**Pull request merged**